plugins/modules/dynamodb_table.py

```
@@ -174,300 +195,737 @@
+    hash_key_type = primary_indexes.get('hash_key_type')
+    range_key_name = primary_indexes.get('range_key_name')
+    range_key_type = primary_indexes.get('range_key_type')
+
+    indexes = list()
+    global_indexes = current_table.get('_global_index_map', {})
+    local_indexes = current_table.get('_local_index_map', {})
+    for index in global_indexes:
+        idx = dict(global_indexes[index])
+        idx.pop('provisioned_throughput', None)
+        indexes.append(idx)
+    for index in local_indexes:
+        idx = dict(local_indexes[index])
+        idx.pop('provisioned_throughput', None)
+        indexes.append(idx)
+
+    compat_results = dict(
+        hash_key_name=hash_key_name,
+        hash_key_type=hash_key_type,
+        range_key_name=range_key_name,
+        range_key_type=range_key_type,
+        indexes=indexes,
+        read_capacity=throughput.get('read_capacity_units', None),
+        region=module.region,
+        table_name=current_table.get('table_name', None),
+        table_status=current_table.get('table_status', None),
+        tags=current_table.get('tags', {}),
+        write_capacity=throughput.get('write_capacity_units', None),
     )
 
+    return compat_results
+
+
+def get_dynamodb_table():
+    table_name = module.params.get('name')
+    try:
+        table = _describe_table(TableName=table_name)
+    except is_boto3_error_code('ResourceNotFoundException'):
+        return None
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:  # pylint: disable=duplicate-except
+        module.fail_json_aws(e, msg='Failed to describe table')
+
+    table = table['Table']
     try:
-        table = Table(table_name, connection=connection)
+        tags = client.list_tags_of_resource(aws_retry=True, ResourceArn=table['TableArn'])['Tags']
+    except is_boto3_error_code('AccessDeniedException'):
+        module.warn('Permission denied when listing tags')
```

Would it be better to fail rather than warn ? other modules seem to fail if it can't get the tags